sql: add explanation for different versions (#2076)
pingcap/docs: reference/sql/statements/use.md

## MySQL compatibility

> For TiDB 2.0, you can use the `USE` statement to access any database. For TiDB 3.0 or later versions, it checks whether you have access to the database when using the `USE` statement. This statement is understood to be fully compatible with MySQL. Any compatibility differences should be [reported via an issue](/report-issue.md) on GitHub.

Reviewer: ```suggestion
In TiDB 2.0 versions, you can use `USE` statements to access any database. In TiDB 3.0 or later versions, TiDB checks whether you have the privilege to access the database when you use `USE` statements. 

Currently, `USE` statements are understood to be fully compatible with MySQL. Any compatibility differences should be [reported via an issue](/report-issue.md) on GitHub.
```
